docs/ops/sequence/CTCLoss_4.md

```diff
@@ -0,0 +1,106 @@
+which is defined independently for each sequence `i`. For `j > last_sequence_symbol`, values in `sequence_mask[j, i]` are all zeros.
+A lenght of target sequence from `labels[i,:]` must not be greater than a lenght of corresponding input sequence. Otherwise, the operation behaves unexpectedly.
+
+*CTCLoss* calculation scheme:
+
+1. Compute probability of `j`-th character at time step `t` for `i`-th input sequence from `inputs` using softmax formula:
+\f[
+p_{t,i,j} = \frac{\exp(inputs[t,i,j])}{\sum^{K}_{k=0}{\exp(inputs[t,i,k])}}
+\f]
+
+2. For a given `i`-th target from `labels[i,:]` find all aligned paths.
+A path `S = (c1,c2,...,cT)` is aligned with a target `G=(g1,g2,...,gT)` if both chains are equal after decoding.
+The decoding removes pads `-1` from a target `G` and merges repeated characters in `G` in case *preprocess_collapse_repeated* equal to True.
+The decoding merges repeated characters in `S` in case *merge_repeated* equal to True and removes blank characters represented by last class index `K`, 
+where `K` is a number of classes including a blank symbol.
+For example, in case default *merge_repeated* and *preprocess_collapse_repeated* a target sequence `(0,3,2,2,-1,-1,-1,-1,-1)` is processed to `(0,3,2,2)` and
+an input `(0,0,4,3,2,2,4,2,4)` is also processed to `(0,3,2,2)`, where `K=4`.
+Compute probabilities of these alignments as follows:
+\f[
+p(S) = \prod_{t=1}^{T} p_{t,i,ct}
+\f]
+
+3. Finally sum up probabilities of all aligned paths for a given target and compute negative logarithm of it:
+\f[
+CTCLoss = -\log \sum_{S} p(S)
+\f]
+
+
+**Attributes**
+
+* *preprocess_collapse_repeated*
+
+  * **Description**: *preprocess_collapse_repeated* is a flag for a preprocessing step before loss calculation, wherein repeated labels passed to the loss are merged into single labels.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: False
+  * **Required**: *no*
+
+* *merge_repeated*
+
+  * **Description**: *merge_repeated* is a flag for merging repeated labels during the CTC loss calculation.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: True
+  * **Required**: *no*
+
+**Inputs**
+
+* **1**: `inputs` - Input tensor with a batch of sequences. Type of elements is *T_F*. Shape of the tensor is `[T, N, C]`, where `T` is the maximum sequence length, `N` is the batch size and `C` is the number of classes. Required.
+
+* **2**: `sequence_mask` - 2D input tensor of type *T_F* with sequence masks for each sequence in the batch. Populated with values 0 and 1. Shape of this input is `[T, N]`. Required.
+
+* **3**: `labels` - 2D tensor with shape `[N, T]` and integer elements containing sequence class indices. A sequence can be shorter than the size `T` of the tensor, all elements that do not code sequence classes are filled with -1. Type of elements is *T_F*, but all values are integers. Required.
```

Comment:
`Type of elements is *T_F*, but all values are integers.` - Why? This input should be a regular i32 or i64 type.

Reply:
Yes, you are correct. `CTCGreedyDecoder` outputs `labels` in floating-point type that is a bug that we need to prevent to be propogated to other operation spec. I will fix it to integer type and insert Cast operation in the transformation. Thanks.

Reply:
Fixed.